tree/ntuple/v7/src/RPageStorageDaos.cxx

```diff
@@ -379,49 +420,48 @@ ROOT::Experimental::Detail::RPageSourceDaos::RPageSourceDaos(std::string_view nt
    auto args = ParseDaosURI(uri);
    auto pool = std::make_shared<RDaosPool>(args.fPoolLabel);
    fDaosContainer = std::make_unique<RDaosContainer>(pool, args.fContainerLabel);
-}
 
+   auto [locator, descBuilder] = RDaosContainerNTupleLocator::LocateNTuple(*fDaosContainer, fNTupleName, *fDecompressor);
 
-ROOT::Experimental::Detail::RPageSourceDaos::~RPageSourceDaos() = default;
+   if (!locator.fAnchor.has_value() || !locator.fAnchor->fNBytesHeader)
```

Comment:
Consider adding the `RDaosContainerNTupleLocator::IsValid()` member function and use it here instead.